File: tool/db_code_generator/Generate/TextWriter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace db_code_generator.Generate
{
    internal class TextWriter
    {
        private readonly StringBuilder _stringBuilder = new();

        public void Write(string text)
        {
            _stringBuilder.Append(text);
        }

        public void WriteLine(string text)
        {
            _stringBuilder.Append($"{text}\r\n");
        }

        public void BreakLine()
        {
            _stringBuilder.Append("\r\n");
        }

        public string Get()
        {
            return _stringBuilder.ToString();
        }
    }
}

File: tool/sox_code_generator/text_writer.cpp
#include "text_writer.h"

namespace cebreiro
{
	void TextWriter::Write(const std::string& text)
	{
		_oss << text;
	}

	void TextWriter::WriteLine(const std::string& text)
	{
		_oss << std::format("{}\r\n", text);
	}

	void TextWriter::BreakLine()
	{
		_oss << "\r\n";
	}

	auto TextWriter::Get() const -> std::string
	{
		return _oss.str();
	}
}
